pyro/infer/abstract_infer.py

```diff
@@ -281,16 +281,18 @@ class TracePredictive(TracePosterior):
     achieved by constraining latent sites to randomly sampled parameter
     values from the model execution traces and running the model forward
     to generate traces with new response ("_RETURN") sites.
-
     :param model: arbitrary Python callable containing Pyro primitives.
-    :param TracePosterior posterior: trace posterior instance holding
-        samples from the model's approximate posterior.
+    :param TracePosterior posterior: trace posterior instance holding samples from the model's approximate posterior.
     :param int num_samples: number of samples to generate.
+    :param keep_sites: The sites which should be sampled from posterior predictive distribution (default: all)
+    :param keep_plates: The plates which should be sampled from posterior predictive distribution (default: all)
```

Comment:
Since we are going to adjusting the plates based on the test data, what is the use case for `keep_plates`?

Reply:
It allows one to drop all sites under plates which are not kept, which is convenient if one has many local sites. Perhaps, I should drop it though, if you find it unimportant 😄

Reply:
I see. I think this argument can seem confusing, so it is best to avoid it. When the API is more mature, and if it turns out to be a pain point, we can consider reinstating it back.